### PR TITLE
Update DG for `update`, `sell` and `restock`

### PR DESCRIPTION
**Pull request by imanamirshah**
TODO: Sequence diagrams for `sell` and `restock`

_Update: sequence diagram completed for all three features_

### CODE REVIEW

**Review by nur-haziq — APPROVED** [review]
LGTM. Sequence diagram seems to abide by the diagram standards.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -281,6 +281,95 @@ and assigns it to `executionUiOutput`.
...
 The ListCommand is concerned only with the execution of the listing operation. It follows a straightforward process that relies on the `ItemList` to format the list of items, ensuring separation of concerns between command execution and UI presentation.
 
+### Sell item
+
+![SellSequenceDiagram](images/SellSequenceDiagram.png)
+
+API: [`SellCommand.java`](https://github.com/AY2324S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/binbash/command/SellCommand.java)
+
+The `sell` command allows users to decrement the quantity of their item as they are being sold off. This is done by
+specifying wither the `name` or `index` to identify the item they want to sell, as well as the `quantity` that they want
```
Comment (YHWong20) [inline docs/DeveloperGuide.md:291]: Note the typo: `wither`

File: docs/images/RestockSequenceDiagram.png
Comment (YHWong20) [inline docs/images/RestockSequenceDiagram.png:1]: Kindly change `RestockItem` to `RestockCommand`.

**Review by YHWong20 — APPROVED** [review]
DG additions are well elaborated on. Kindly help to update the table of contents as well. Thanks!

**Pull request merged**